app/scripts/lib/rpc-method-middleware/handlers/request-accounts.js

```diff
@@ -0,0 +1,103 @@
+import { ethErrors } from 'eth-rpc-errors';
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+
+/**
+ * This method attempts to retrieve the Ethereum accounts available to the
+ * requester, or initiate a request for account access if none are currently
+ * available. It is essentially a wrapper of wallet_requestPermissions that
+ * only errors if the user rejects the request. We maintain the method for
+ * backwards compatibility reasons.
+ */
+
+const requestEthereumAccounts = {
+  methodNames: [MESSAGE_TYPE.ETH_REQUEST_ACCOUNTS],
+  implementation: requestEthereumAccountsHandler,
+  hookNames: {
+    origin: true,
+    getAccounts: true,
+    getUnlockPromise: true,
+    hasPermission: true,
+    requestAccountsPermission: true,
+  },
+};
+export default requestEthereumAccounts;
+
+// Used to rate-limit pending requests to one per origin
+const locks = new Set();
+
+/**
+ * @typedef {Record<string, string | Function>} RequestEthereumAccountsOptions
+ * @property {string} origin - The requesting origin.
+ * @property {Function} getAccounts - Gets the accounts for the requesting
+ * origin.
+ * @property {Function} getUnlockPromise - Gets a promise that resolves when
+ * the extension unlocks.
+ * @property {Function} hasPermission - Returns whether the requesting origin
+ * has the specified permission.
+ * @property {Function} requestAccountsPermission - Requests the `eth_accounts`
+ * permission for the requesting origin.
+ */
+
+/**
+ *
+ * @param {import('json-rpc-engine').JsonRpcRequest<unknown>} req - The JSON-RPC request object.
```

Comment:
The parameter has a leading underscore
```suggestion
 * @param {import('json-rpc-engine').JsonRpcRequest<unknown>} _req - The JSON-RPC request object.
```

Reply:
Fixed in: d49f868ae